### ISSUE TITLE
internal_listener config can't run (panic: unset oneof)

### ISSUE THREAD

**Issue by pxpnetworks** [envoyproxy/envoy#23973]
Hi, i have an issue running envoy config with internal_listener...
Tried with [this official](https://github.com/envoyproxy/envoy/blob/ff49762696b2e6ed3f408a22e1f7a1b7d2487318/configs/terminate_http_in_http2_connect.yaml) config and it still fails with the same error 
```
[2022-11-14 07:31:37.407][31291][critical][assert] [source/common/network/utility.cc:491] panic: unset oneof
[2022-11-14 07:31:37.407][31291][critical][backtrace] [./source/server/backtrace.h:104] Caught Aborted, suspect faulting address 0x7a3b
[2022-11-14 07:31:37.407][31291][critical][backtrace] [./source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
[2022-11-14 07:31:37.407][31291][critical][backtrace] [./source/server/backtrace.h:92] Envoy version: edd69583372955fdfa0b8ca3820dd7312c094e46/1.23.1/Clean/RELEASE/BoringSSL
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:96] #0: __restore_rt [0x7f6eb1409140]
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:98] #1: [0x5590c2f6a28e]
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:98] #2: [0x5590c2f81345]
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:98] #3: [0x5590c2f80146]
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:98] #4: [0x5590c2fd8c6f]
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:98] #5: [0x5590c2c0e346]
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:98] #6: [0x5590c2c09989]
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:98] #7: [0x5590c16e6291]
[2022-11-14 07:31:37.408][31291][critical][backtrace] [./source/server/backtrace.h:98] #8: [0x5590c16e574d]
[2022-11-14 07:31:37.409][31291][critical][backtrace] [./source/server/backtrace.h:98] #9: [0x5590c16e6b4b]
[2022-11-14 07:31:37.409][31291][critical][backtrace] [./source/server/backtrace.h:98] #10: [0x5590c16e6ccc]
[2022-11-14 07:31:37.409][31291][critical][backtrace] [./source/server/backtrace.h:98] #11: [0x5590c16e311c]
[2022-11-14 07:31:37.409][31291][critical][backtrace] [./source/server/backtrace.h:96] #12: __libc_start_main [0x7f6eb106eac5]
```
This is tested with envoy 1.23.0 and 1.23.1 and results are the same.
Running the binary on Oracle Linux 8.6 Kernel 5.4.17

Can anyone please help me on this? Could be something simple but i cant get my head around what panic: unset oneof means..

BR,
Stoyan

**Comment by liverbirdkte** [comment]
I also met problems with internal listeners on Envoy 1.23, although I forgot the exact error encountered, seems related to a configuration check.

When I switched to Envoy 1.24, everything works fine.  Does 1.24 work for you?

**Comment by pxpnetworks** [comment]
Yeah, i confirm... on 1.24.0 it works.
My issue is some of the VMs i run it use glibc <2.30 which is needed for v1.24.0 to run :)

BR,
Stoyan

**Comment by zuercher** [comment]
In the future, please report crashing bugs to the envoy-security@googlegroups.com mailing list so that it can be appropriately triaged before public release. In this case, the crash appears to be caused by using a feature that was not fully implemented in v1.23. Since the crash is triggered by operator-controlled configuration, this issue doesn't need an embargo.

Internal listener support was introduced in v1.24, so I'm going to close this as won't fix.
